api/v1alpha3/azurecluster_webhook.go

```diff
@@ -56,8 +60,30 @@ func (c *AzureCluster) ValidateCreate() error {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) error {
 	clusterlog.Info("validate update", "name", c.Name)
+	var allErrs field.ErrorList
 	old := oldRaw.(*AzureCluster)
-	return c.validateCluster(old)
```

Comment:
we still want to run `validateCluster` to make sure that the fields that did get changed are still valid